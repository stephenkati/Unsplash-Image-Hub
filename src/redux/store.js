import { configureStore } from '@reduxjs/toolkit';
import DetailssSlice from './DetailsSlice';
import CollectionsReducer from './CollectionsSlice';

const store = configureStore({
  reducer: {
    collectionDetails: DetailssSlice,
    collections: CollectionsReducer,
  },
});

export default store;
