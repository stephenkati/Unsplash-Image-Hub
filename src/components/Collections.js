import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { MdOutlineArrowCircleRight } from 'react-icons/md';
import { fetchCollections } from '../redux/CollectionsSlice';
import unsplashImage from '../image/shinny-hand.png';

const Collections = () => {
  const dispatch = useDispatch();
  const [search, setSearch] = useState('');
  const [page, setPage] = useState(1);

  const handlePrevious = () => setPage(page - 1);
  const handleNext = () => setPage(page + 1);

  useEffect(() => {
    dispatch(fetchCollections(page));
  }, [dispatch, page]);

  const collection = useSelector((state) => state.collections.collections);

  return (
    <div className="collection-container">
      <div className="search">
        <form>
          <input
            type="text"
            placeholder="Search collection name"
            onChange={(e) => setSearch(e.target.value)}
          />
        </form>
      </div>
      <div className="collection-header">
        <img
          src={unsplashImage}
          alt="unsplash"
        />
        <h1 className="header-title">Unsplash Collections</h1>
      </div>
      <div className="stats-by-collection">
        <h2>Stats by Collection</h2>
      </div>
      <div className="collection-list">
        {
          collection
            .filter((collection) => (search.toLowerCase() === ''
              ? collection
              : collection.title.toLowerCase().includes(search)))
            .map((collection) => (
              <NavLink
                to={`/details/${encodeURIComponent(collection.title)}`}
                className="collection-link"
                key={collection.id}
              >
                <div
                  role="button"
                  tabIndex={0}
                  className="collection"
                >
                  <MdOutlineArrowCircleRight
                    className="arrow"
                  />
                  <img
                    src={collection.cover_photo.urls.small}
                    alt={collection.title}
                    className="collection-img"
                    loading="lazy"
                  />
                  <div className="collection-title">
                    {collection.title}
                  </div>
                  <div className="cover-likes">
                    {collection.total_photos}
                  </div>
                </div>
              </NavLink>
            ))
        }
      </div>
      <div className="page">
        {page > 1 && (
          <span
            role="button"
            tabIndex={0}
            onClick={handlePrevious}
            onKeyDown={handlePrevious}
          >
            Prev
          </span>
        )}
        {page}
        <span
          role="button"
          tabIndex={0}
          onClick={handleNext}
          onKeyDown={handleNext}
        >
          Next
        </span>
      </div>
    </div>
  );
};

export default Collections;
