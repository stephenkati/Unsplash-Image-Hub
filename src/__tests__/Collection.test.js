import { render } from '@testing-library/react';
import Collections from '../components/Collections';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import store from '../redux/store';

describe('tests for collections', () => {

  test('snapshot for collection file', ()=> {

    const { component } = render(
        <Provider store={store}>
          <Collections />
        </Provider>
    );

      expect(component).toMatchSnapshot();
  });
});
