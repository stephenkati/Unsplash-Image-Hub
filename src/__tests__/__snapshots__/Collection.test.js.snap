// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for collections snapshot for collection file 1`] = `undefined`;
